libs/core/src/components/pds-select/test/pds-select.spec.tsx

```diff
@@ -0,0 +1,584 @@
+
+    expect(page.root).toEqualHtml(`
+      <pds-select autocomplete="off" component-id="field-1" label="Name">
+        <mock:shadow-root>
+          <div class="pds-select">
+            <label htmlFor="field-1">Name</label>
+            <select autocomplete="off" class="pds-select__field" id="field-1"></select>
+            <div aria-hidden="true" class="hidden">
+              <slot></slot>
+            </div>
+            <pds-icon class="pds-select__select-icon" name="enlarge"></pds-icon>
+          </div>
+        </mock:shadow-root>
+      </pds-select>
+    `);
+  });
+
+  it('renders with autocomplete attribute set', async () => {
+    const page = await newSpecPage({
+      components: [PdsSelect],
+      html: `<pds-select autocomplete="on" component-id="field-1" label="Name"></pds-select>`,
+    });
+
+    expect(page.root).toEqualHtml(`
+      <pds-select autocomplete="on" component-id="field-1" label="Name">
+        <mock:shadow-root>
+          <div class="pds-select">
+            <label htmlFor="field-1">Name</label>
+            <select autocomplete="on" class="pds-select__field" id="field-1"></select>
+            <div aria-hidden="true" class="hidden">
+              <slot></slot>
+            </div>
+            <pds-icon class="pds-select__select-icon" name="enlarge"></pds-icon>
+          </div>
+        </mock:shadow-root>
+      </pds-select>
+    `);
+  });
+
+  it('renders with the multiple attribute', async () => {
+    const { root } = await newSpecPage({
+      components: [PdsSelect],
+      html: `<pds-select component-id="field-1" multiple></pds-select>`,
+    });
+
+    expect(root).toEqualHtml(`
+      <pds-select component-id="field-1" multiple="">
+        <mock:shadow-root>
+          <div class="pds-select">
+            <label htmlFor="field-1"></label>
+            <select class="pds-select__field" id="field-1" multiple="">
+            </select>
+            <div aria-hidden="true" class="hidden">
+              <slot></slot>
+            </div>
+          </div>
+        </mock:shadow-root>
+      </pds-select>
+    `);
+  });
+
+  it('calls updateSelectedOption when value changes with multiple values', async () => {
+    const page = await newSpecPage({
+      components: [PdsSelect],
+      html: `<pds-select component-id="field-1" multiple></pds-select>`,
+    });
+
+    const component = page.rootInstance;
+    const updateSelectedOptionSpy = jest.spyOn(component, 'updateSelectedOption');
+
+    // Mock the selectEl and its options property
+    const mockOption1 = { value: 'option1', selected: false } as HTMLOptionElement;
+    const mockOption2 = { value: 'option2', selected: false } as HTMLOptionElement;
+    const mockOption3 = { value: 'option3', selected: false } as HTMLOptionElement;
+    const mockOptions = [mockOption1, mockOption2, mockOption3];
+    const mockSelectEl = {
+      options: mockOptions,
+    } as unknown as HTMLSelectElement;
+    component.selectEl = mockSelectEl;
```

Comment:
Non-blocker but a thought: a lot of these tests require setting up mock options. I wonder if there's a way we can set these up outside of the test and pull them in as necessary so we don't have to repeat them.

Reply:
This came up in a convo with Julian as well. I think these tests will need a refactor but when I tried updating them I kept hitting other issues so will save for a follow up.